cmd/ham_server/main.go

```diff
@@ -32,11 +32,16 @@ func main() {
 	server = restapi.NewServer(api)
 	server.Port = 8080
 
-	defer server.Shutdown()
+	defer func() {
+		if err := server.Shutdown(); err != nil {
+			log.Printf("Error shutting down the server: %s", err)
+		}
+	}()
 
 	server.ConfigureAPI()
 
 	if err := server.Serve(); err != nil {
-		log.Fatalln(err)
+		log.Println(err)
+		panic("Error occurred")
 	}
 }
```

Comment:
`ham_server` in `cmd/ham_server/main.go` is not correct. 
should be meaningful and relevant name, like: `cmd/chat_server/main.go` or `cmd/chat_app_server/main.go`

Reply:
Changed